src/main/scala/pipedsl/codegen/bsv/BSVSyntax.scala

```diff
@@ -80,7 +80,7 @@ object BSVSyntax {
           mtyp.tparams.find(bv => bv.name == bsints.reqIdName).get.typ
         }
         getLockedMemType(mem, mtyp, lidtyp, limpl, useTypeVars = false, None)
-      case TSizedInt(len, unsigned) => BSizedInt(unsigned, len)
+      case TSizedInt(len, unsigned) => BSizedInt(unsigned.unsigned(), len.asInstanceOf[TBitWidthLen].len)
```

Comment:
Hmmm it would be nice to not have these casts in here.

One paradigm is designing a partial function that lets us write matching nicely (see `matchOrError` in Syntax.scala).

The other option is something more complicated that involves separate representations for the type var and the type. This one seems overkill to me.